templates/source/base_source.html

```diff
@@ -39,5 +39,11 @@ <h2><i class="fa fa-fw fa-paperclip"></i> {% trans "Add source" %}</h2>
 
 {% block extra_js %}
     {% include 'partials/source_edit_js.html' %}
-{% endblock %}
+    <script>
+
+      $('button[name="cancel-button"]').on('click', function() {
+        window.history.back()
+      });
```

Comment:
Using JavaScript for this redirect doesn't quite give the behavior I expect -- e.g. if I view the access points for this source, then switch back to the `Basics` tab, then hit `Cancel`, I'll be taken back to the `Access points` page when I would expect to be taken back to the detail view for the source.

Instead, I think it'd make for more consistent behavior if we turned the `Cancel` buttons into links that took you back to the detail page for the source in question.

Reply:
sounds good!

Reply:
what do you think about this JS behavior for cancelling source creation? (versus access points)

Reply:
Good Q! I notice now that Tom's instructions in the PR  specified that it should 
"either return the user to the page they were looking at before, or take them home" so I see why the JS behavior seemed appropriate.

To me, the "typical" behavior of cancel buttons would indicate the following actions:

| action | cancel redirects to |
| --- | --- |
| Source create | Source list |
| Source edit | Source detail |
| Access point edit | Access point detail |

Tom may have different thoughts, but we can talk that over when he returns from vacation. For now, do these redirects seem feasible? They may require adjusting the template so that we can pass the redirect value in dynamically.

Reply:
yep, i can do these redirects. thanks jean!